QuizApp/App/Screens/Quiz details/QuizDetailsViewController.swift

```diff
@@ -0,0 +1,167 @@
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        createViews()
+        styleViews()
+        defineLayoutForViews()
+        setupNavigationBar()
+        bindViewModel()
+    }
+
+}
+
+extension QuizDetailsViewController: ConstructViewsProtocol {
+
+    func createViews() {
+        leaderboardButton = UIButton()
+        view.addSubview(leaderboardButton)
+
+        quizContainterView = UIView()
+        view.addSubview(quizContainterView)
+
+        titleLabel = UILabel()
+        quizContainterView.addSubview(titleLabel)
+
+        descriptionLabel = UILabel()
+        quizContainterView.addSubview(descriptionLabel)
+
+        imageView = UIImageView()
+        quizContainterView.addSubview(imageView)
+
+        startButton = UIButton()
+        quizContainterView.addSubview(startButton)
+    }
+
+    func styleViews() {
+        view.applyGradientWith([UIColor.loginBackgroundTop.cgColor, UIColor.loginBackgroundBottom.cgColor])
+
+        let leaderboardTitle = NSAttributedString(
+            string: "Leaderboard",
+            attributes: [
+                NSAttributedString.Key.foregroundColor: UIColor.white,
+                NSAttributedString.Key.underlineStyle: NSUnderlineStyle.single.rawValue,
+                NSAttributedString.Key.font: Fonts.sourceSansProBold18.font]
+        )
+        leaderboardButton.setAttributedTitle(leaderboardTitle, for: .normal)
+        leaderboardButton.addTarget(self, action: #selector(leaderboardButtonTapped), for: .touchUpInside)
+
+        quizContainterView.backgroundColor = .white.withAlphaComponent(0.3)
+        quizContainterView.layer.cornerRadius = 10
+
+        titleLabel.style(with: nil, color: .white, alignment: .center, font: Fonts.sourceSansProBold32.font)
+
+        descriptionLabel.style(with: nil, color: .white, alignment: .center, font: Fonts.sourceSansProSemiBold18.font)
+        descriptionLabel.numberOfLines = 0
+
+        imageView.style(with: nil, contentMode: .scaleAspectFit, radius: 10)
+
+        startButton.setTitle("Start Quiz", for: .normal)
+        startButton.setTitleColor(.loginPurple, for: .normal)
+        startButton.titleLabel?.font = Fonts.sourceSansProBold16.font
+        startButton.layer.cornerRadius = 22
+        startButton.backgroundColor = .white
+        startButton.addTarget(self, action: #selector(startButtonTapped), for: .touchUpInside)
+    }
+
+    func defineLayoutForViews() {
+        leaderboardButton.snp.makeConstraints {
+            $0.top.equalTo(view.safeAreaLayoutGuide).offset(Constants.horizontalMargin)
+            $0.trailing.equalTo(view.snp.trailing).inset(Constants.horizontalMargin)
+            $0.width.height.equalTo(150)
+        }
+
+        quizContainterView.snp.makeConstraints {
+            $0.centerX.centerY.equalTo(view.safeAreaLayoutGuide)
+            $0.height.equalTo(400)
+            $0.width.equalTo(view.snp.width).inset(20)
```

Comment:
Why set insets on width? 
Setting .leading and .trailing is more readable, and consistent.

Reply:
Also, `horizontalMargin`.